ui/src/message_popup/transformations.jsx

```diff
@@ -0,0 +1,33 @@
+import _ from 'lodash';
+import {allStudents} from '../data/virtual_school.js';
+import {learningObjectives} from '../data/learning_objectives.js';
+import {allQuestions} from './questions.js';
+
+
+export function withStudents(questions) {
+  return questions.map((question) => {
+    const student = _.find(allStudents, {id: question.studentId });
+    return _.extend({student}, question);
+  });
+}
+
+export function withLearningObjective(question) {
+  const learningObjective = _.find(learningObjectives, { id: question.learningObjectiveId });
+  return {
+    ...question,
+    learningObjective
+  };
+}
+
+export function questionsForCompetencies(competencyGroup) {
+  const withCompetencyGroups = _.compact(allQuestions.map((question) => {
+    const learningObjective = _.find(learningObjectives, { id: question.learningObjectiveId });
+    if (learningObjective.competencyGroup !== competencyGroup) return null;
+    return {
+      ...question,
+      competencyGroup: learningObjective.competencyGroup
+    };
+  }));
+
+  return _.shuffle(withStudents(withCompetencyGroups));
+}
```

Comment:
I am about to make a commit to my open pull request that moved these to the inside of `instructions_card.jsx` but I think that this is a much better idea so we should just make sure to change that when merging whichever one is merged last.